org.eclipse.lsp4e/src/org/eclipse/lsp4e/LanguageClientImpl.java

```diff
@@ -156,4 +156,11 @@ public CompletableFuture<ShowDocumentResult> showDocument(ShowDocumentParams par
 			return new ShowDocumentResult(true);
 		});
 	}
+
+	/**
+	 * Dispose language client.
+	 */
+	public void dispose() {
+		progressManager.dispose();
```

Comment:
Nice find! This could also fix an issue I am sometimes seeing with the Dart Language Server where endlessly running progress bars in the progress view pile up. Would be nice if this could make it into the upcoming release.

Reply:
To be honnest with you, I have not tested andI did that just to find a usecase with dispose in LSP4E. In my case, my language server which implements LanguageClientImpl (in IJ) override dispose method to remove some project listeners added in the constructor.

Reply:
I have taken over this part of the PR in https://github.com/eclipse/lsp4e/pull/705 and plan to work on it.

Reply:
Thanks @rubenporras !